truss/tests/patch/test_calc_patch.py

```diff
@@ -316,23 +315,15 @@ def config_op(config: TrussConfig):
         filename = "requirements.txt"
         with (custom_model_truss_dir / filename).open("w") as req_file:
             req_file.write(requirements_contents)
-        config.requirements_file = filename
-        config.requirements.clear()
 
     patches = _apply_config_change_and_calc_patches(
         custom_model_truss_dir,
         config_op=config_op,
         config_pre_op=pre_config_op,
     )
-    assert len(patches) == 4
+    assert len(patches) == 3
```

Comment:
do we have a test where we remove the requirements file and add new requirements in config.yaml?

Reply:
yeah, we have it 

```
def test_calc_truss_patch_handles_requirements_file_removal(
    custom_model_truss_dir: Path,
):
    requirements_contents = """xformers"""
    filename = "requirements.txt"

    def pre_config_op(config: TrussConfig):
        with (custom_model_truss_dir / filename).open("w") as req_file:
            req_file.write(requirements_contents)
        config.requirements_file = filename
        config.requirements.clear()

    def config_op(config: TrussConfig):
        requirements_contents = ["xformers", "requests"]
        config.requirements.extend(requirements_contents)
        config.requirements_file = ""
        os.remove(custom_model_truss_dir / filename)

    patches = _apply_config_change_and_calc_patches(
        custom_model_truss_dir,
        config_op=config_op,
        config_pre_op=pre_config_op,
    )
    assert len(patches) == 2
    assert patches == [
        Patch(
            type=PatchType.CONFIG,
            body=ConfigPatch(
                action=Action.UPDATE,
                config=yaml.safe_load((custom_model_truss_dir / "config.yaml").open()),
            ),
        ),
        Patch(
            type=PatchType.PYTHON_REQUIREMENT,
            body=PythonRequirementPatch(
                action=Action.ADD,
                requirement="requests",
            ),
        ),
    ]
```

Reply:
updated in this PR so that we're adding requirements when doing this